server/src/main/java/info/doseamigos/authusers/AccessTokenDeserializer.java

```diff
@@ -0,0 +1,53 @@
+package info.doseamigos.authusers;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Jackson Deserializer for AccessToken to turn into AuthUsers.  If we pass just the access token in, then we
+ */
+public class AccessTokenDeserializer extends JsonDeserializer<AuthUser> {
```

Comment:
I couldn't think of a better way of handling this.  It sucks that I can't inject in stuff since it's used in annotations, but it does work and do it's job.